azkaban-common/src/test/java/azkaban/executor/RunningExecutionsUpdaterTest.java

```diff
@@ -0,0 +1,153 @@
+package azkaban.executor;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+import azkaban.alert.Alerter;
+import azkaban.metrics.CommonMetrics;
+import azkaban.utils.Pair;
+import com.google.common.collect.ImmutableMap;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.joda.time.DateTimeUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class RunningExecutionsUpdaterTest {
+
+  private static final int EXECUTION_ID_77 = 77;
+  private static final ExecutorManagerException API_CALL_EXCEPTION =
+      new ExecutorManagerException("Mocked API timeout");
+
+  @Mock
+  ExecutorManagerUpdaterStage updaterStage;
+  @Mock
+  AlerterHolder alerterHolder;
+  @Mock
+  CommonMetrics commonMetrics;
+  @Mock
+  ExecutorApiGateway apiGateway;
+  @Mock
+  ExecutionFinalizer executionFinalizer;
+  @Mock
+  private Alerter mailAlerter;
+
+  private ExecutableFlow execution;
+  private RunningExecutions runningExecutions;
+  private Executor activeExecutor;
+
+  private RunningExecutionsUpdater updater;
+
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+    this.execution = new ExecutableFlow();
+    this.execution.setExecutionId(EXECUTION_ID_77);
+    this.activeExecutor = new Executor(1, "activeExecutor-1", 9999, true);
+    this.runningExecutions = new RunningExecutions();
+    this.runningExecutions.get().put(EXECUTION_ID_77, new Pair<>(
+        new ExecutionReference(EXECUTION_ID_77, this.activeExecutor), this.execution));
+    this.updater = new RunningExecutionsUpdater(this.updaterStage, this.alerterHolder,
+        this.commonMetrics, this.apiGateway, this.runningExecutions, this.executionFinalizer);
+    when(this.alerterHolder.get("email")).thenReturn(this.mailAlerter);
+  }
+
+  @After
+  public void tearDown() {
+    DateTimeUtils.setCurrentMillisSystem();
+  }
+
+  @Test
+  public void updateExecutionsStillRunning() throws Exception {
```

Comment:
Consider adding the test for finished executions as well?

Reply:
Yes, somehow didn't occur to me.